File: cpp/src/ByteBuffer.cpp


#include "ByteBuffer.h"

void ByteBuffer::append(int i) {
    _buffer->push_back(i);
}

int ByteBuffer::getAvailable() {

    return _buffer->size()  - _current;
}

int ByteBuffer::read() {
    int value = _buffer->at(_current);
    _current += 1;
    return value;
}

ByteBuffer::ByteBuffer() {
   _buffer = new std::vector<int>();
   _current = 0;
}

File: java/src/main/java/org/codingdojo/kata/encode/ByteBuffer.java
package org.codingdojo.kata.encode;

import java.util.ArrayList;
import java.util.List;

public class ByteBuffer {

    private final List<Byte> buffer = new ArrayList<>();
    private int current;

    public void append(int i) {
        buffer.add((byte) i);
    }

    public int read() {
        return buffer.get(current++);
    }

    public int getAvailable() {
        return buffer.size() - current;
    }
}
